sonic_data_client/db_client.go

```diff
@@ -663,91 +730,103 @@ func enqueFatalMsg(c *DbClient, msg string) {
+				continue
+			}
+
+			path2ValueMap[tblPath] = val
+			fv := map[string]string{tblPath.jsonField: val}
+			msi[tblPath.jsonTableKey] = fv
+			log.V(6).Infof("new value %v for %v", val, tblPath)
+		}
+
+		return msi
 	}
-	synced := bool(false)
+
+	sendVal := func(msi map[string]interface{}) error {
+		val, err := msi2TypedValue(msi)
+		if err != nil {
+			enqueFatalMsg(c, err.Error())
+			return err
+		}
+
+		spbv := &spb.Value{
+			Prefix:    c.prefix,
+			Path:      gnmiPath,
+			Timestamp: time.Now().UnixNano(),
+			Val:       val,
+		}
+
+		if err = c.q.Put(Value{spbv}); err != nil {
+			log.V(1).Infof("Queue error:  %v", err)
+			return err
+		}
+
+		return nil
+	}
+
+	msi := readVal()
+	if err := sendVal(msi); err != nil {
+		c.synced.Done()
+		return
+	}
+	c.synced.Done()
 
 	for {
 		select {
 		case <-c.channel:
 			log.V(1).Infof("Stopping dbFieldMultiSubscribe routine for Client %s ", c)
 			return
-		default:
-			msi := make(map[string]interface{})
-			for _, tblPath := range tblPaths {
-				var key string
-				if tblPath.tableKey != "" {
-					key = tblPath.tableName + tblPath.delimitor + tblPath.tableKey
-				} else {
-					key = tblPath.tableName
-				}
-				// run redis get directly for field value
-				redisDb := Target2RedisDb[tblPath.dbName]
-				val, err := redisDb.HGet(key, tblPath.field).Result()
-				if err == redis.Nil {
-					if tblPath.jsonField != "" {
-						// ignore non-existing field which was derived from virtual path
-						continue
-					}
-					log.V(2).Infof("%v doesn't exist with key %v in db", tblPath.field, key)
-					enqueFatalMsg(c, fmt.Sprintf("%v doesn't exist with key %v in db", tblPath.field, key))
-					return
-				}
-				if err != nil {
-					log.V(1).Infof(" redis HGet error on %v with key %v", tblPath.field, key)
-					enqueFatalMsg(c, fmt.Sprintf(" redis HGet error on %v with key %v", tblPath.field, key))
-					return
-				}
-				if val == path2ValueMap[tblPath] {
-					continue
-				}
-				path2ValueMap[tblPath] = val
-				fv := map[string]string{tblPath.jsonField: val}
-				msi[tblPath.jsonTableKey] = fv
-				log.V(6).Infof("new value %v for %v", val, tblPath)
-			}
+		case <-IntervalTicker(interval):
+			msi := readVal()
 
 			if len(msi) != 0 {
-				val, err := msi2TypedValue(msi)
-				if err != nil {
-					enqueFatalMsg(c, err.Error())
-					return
-				}
-
-				spbv := &spb.Value{
-					Prefix:    c.prefix,
-					Path:      gnmiPath,
-					Timestamp: time.Now().UnixNano(),
-					Val:       val,
-				}
-
-				if err = c.q.Put(Value{spbv}); err != nil {
+				if err := sendVal(msi); err != nil {
 					log.V(1).Infof("Queue error:  %v", err)
 					return
 				}
-
-				if !synced {
-					c.synced.Done()
-					synced = true
-				}
 			}
-			// check again after 200 millisends, to use configured variable
-			time.Sleep(time.Millisecond * 200)
 		}
 	}
 }
 
 // for subscribe request with granularity of table field, the value is fetched periodically.
```

Comment:
this comment needs update to cover the sampling mode behavior

Reply:
same comment for the above function.

Reply:
Updated the doc for `dbFieldSubscribe` and `dbFieldMultiSubscribe`